packages/@aws-cdk/aws-logs/README.md

```diff
@@ -302,6 +302,19 @@ const pattern = logs.FilterPattern.spaceDelimited('time', 'component', '...', 'r
   .whereNumber('result_code', '!=', 200);
 ```
 
+## Logs Insights Query Definition
+
+Creates a query definition for CloudWatch Logs Insights.
+
+Example:
+
+```ts
+new logs.QueryDefinition(this, 'QueryDefinition', {
+  queryDefinitionName: 'MyQuery',
+  queryString: 'fields @timestamp, @message | sort @timestamp desc | limit 20',
```

Comment:
This query string can be pretty sophisticated: https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CWL_QuerySyntax.html

Could we create a type for this besides just string? This API could be similar to the one we offer for IAM policies, where users can call methods to add different fields to it (but here, they would call methods to add different commands (parse, limit, sort, stats, etc)